EFCore.CheckConstraints/Internal/EnumCheckConstraintConvention.cs

```diff
@@ -1,25 +1,44 @@
 using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Reflection;
 using System.Text;
 using Microsoft.EntityFrameworkCore;
 using Microsoft.EntityFrameworkCore.Metadata;
 using Microsoft.EntityFrameworkCore.Metadata.Builders;
 using Microsoft.EntityFrameworkCore.Metadata.Conventions;
-using Microsoft.EntityFrameworkCore.Metadata.Conventions.Infrastructure;
 using Microsoft.EntityFrameworkCore.Storage;
+using Microsoft.EntityFrameworkCore.Storage.ValueConversion;
 
 namespace EFCore.CheckConstraints.Internal;
 
 public class EnumCheckConstraintConvention : IModelFinalizingConvention
 {
     private readonly IRelationalTypeMappingSource _typeMappingSource;
     private readonly ISqlGenerationHelper _sqlGenerationHelper;
+    private readonly IReadOnlyDictionary<Type, MethodInfo> _supportedEnumValueTypes;
 
-    public EnumCheckConstraintConvention(
-        IRelationalTypeMappingSource typeMappingSource,
-        ISqlGenerationHelper sqlGenerationHelper)
+    public EnumCheckConstraintConvention(IRelationalTypeMappingSource typeMappingSource, ISqlGenerationHelper sqlGenerationHelper)
     {
         _typeMappingSource = typeMappingSource;
         _sqlGenerationHelper = sqlGenerationHelper;
+
+        var method = GetType().GetTypeInfo().GetDeclaredMethod(nameof(TryGetMinMax))!;
+
+        _supportedEnumValueTypes = new Dictionary<Type, MethodInfo>
```

Comment:
Not terribly important, but given that enums with various underlying types are pretty rare, I think it makes sense to just use this dictionary as a cache (i.e. call MakeGenericMethod the first time it's needed, cache the result).

I admit this plugin isn't perf-sensitive, but it still seems a bit better.

Reply:
Okay, can do. I'll still cache the method, but then only generate when requested.